From 67f8fa27f8a6b4b12b4b3b48da8139651ed2e620 Mon Sep 17 00:00:00 2001
From: Lionel Debieve <lionel.debieve@foss.st.com>
Date: Tue, 30 Nov 2021 23:59:14 +0100
Subject: [PATCH 0223/1695] dt-bindings: add platform specific ETZPC bindings

Define ETZPC bindings for STM32MP15 and STM32MP13 and add these
header files into the stm32mp_dt_bindings helper.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 3813dbba953db3221457b98fbf5c5cc6d1f6c00d dt-bindings: add platform specific ETZPC bindings]
Change-Id: I91451738732d879ffa65f1384f83e5aaeda61e88
Signed-off-by: Lionel Debieve <lionel.debieve@foss.st.com>
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
---
 core/arch/arm/plat-stm32mp1/main.c            |  1 -
 .../arch/arm/plat-stm32mp1/shared_resources.c |  1 -
 core/include/drivers/stm32mp_dt_bindings.h    |  4 +-
 .../include/dt-bindings/soc/stm32mp13-etzpc.h | 78 +++++++++++++++++
 .../dt-bindings/soc/stm32mp15-etzpc.h}        | 85 +++++--------------
 5 files changed, 104 insertions(+), 65 deletions(-)
 create mode 100644 core/include/dt-bindings/soc/stm32mp13-etzpc.h
 rename core/{arch/arm/plat-stm32mp1/drivers/stm32mp1_etzpc.h => include/dt-bindings/soc/stm32mp15-etzpc.h} (60%)

diff --git a/core/arch/arm/plat-stm32mp1/main.c b/core/arch/arm/plat-stm32mp1/main.c
index d99a7fb13..5948bc55c 100644
--- a/core/arch/arm/plat-stm32mp1/main.c
+++ b/core/arch/arm/plat-stm32mp1/main.c
@@ -14,7 +14,6 @@
 #include <drivers/stm32_iwdg.h>
 #include <drivers/stm32_tamp.h>
 #include <drivers/stm32_uart.h>
-#include <drivers/stm32mp1_etzpc.h>
 #include <drivers/stm32mp_dt_bindings.h>
 #include <io.h>
 #include <kernel/boot.h>
diff --git a/core/arch/arm/plat-stm32mp1/shared_resources.c b/core/arch/arm/plat-stm32mp1/shared_resources.c
index d51ec9766..6f056c287 100644
--- a/core/arch/arm/plat-stm32mp1/shared_resources.c
+++ b/core/arch/arm/plat-stm32mp1/shared_resources.c
@@ -7,7 +7,6 @@
 #include <drivers/pinctrl.h>
 #include <drivers/stm32_etzpc.h>
 #include <drivers/stm32_gpio.h>
-#include <drivers/stm32mp1_etzpc.h>
 #include <drivers/stm32mp1_rcc.h>
 #include <drivers/stm32mp_dt_bindings.h>
 #include <initcall.h>
diff --git a/core/include/drivers/stm32mp_dt_bindings.h b/core/include/drivers/stm32mp_dt_bindings.h
index b39dae525..4596df4e8 100644
--- a/core/include/drivers/stm32mp_dt_bindings.h
+++ b/core/include/drivers/stm32mp_dt_bindings.h
@@ -1,6 +1,6 @@
 /* SPDX-License-Identifier: BSD-2-Clause */
 /*
- * Copyright (c) 2022, STMicroelectronics
+ * Copyright (c) 2017-2022, STMicroelectronics
  */
 #ifndef __DRIVERS_STM32MP_DT_BINDINGS_H
 #define __DRIVERS_STM32MP_DT_BINDINGS_H
@@ -10,6 +10,7 @@
 #include <dt-bindings/clock/stm32mp13-clksrc.h>
 #include <dt-bindings/regulator/st,stm32mp13-regulator.h>
 #include <dt-bindings/reset/stm32mp13-resets.h>
+#include <dt-bindings/soc/stm32mp13-etzpc.h>
 #include <dt-bindings/soc/stm32mp13-tzc400.h>
 #endif
 
@@ -18,6 +19,7 @@
 #include <dt-bindings/clock/stm32mp1-clksrc.h>
 #include <dt-bindings/regulator/st,stm32mp15-regulator.h>
 #include <dt-bindings/reset/stm32mp1-resets.h>
+#include <dt-bindings/soc/stm32mp15-etzpc.h>
 #include <dt-bindings/soc/stm32mp15-tzc400.h>
 #endif
 
diff --git a/core/include/dt-bindings/soc/stm32mp13-etzpc.h b/core/include/dt-bindings/soc/stm32mp13-etzpc.h
new file mode 100644
index 000000000..0afd71c72
--- /dev/null
+++ b/core/include/dt-bindings/soc/stm32mp13-etzpc.h
@@ -0,0 +1,78 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright (c) 2022-2024, STMicroelectronics
+ */
+
+#ifndef _DT_BINDINGS_SOC_STM32MP13_ETZPC_H
+#define _DT_BINDINGS_SOC_STM32MP13_ETZPC_H
+
+/*  define DECPROT modes */
+#define DECPROT_S_RW		0x0
+#define DECPROT_NS_R_S_W	0x1
+#define DECPROT_NS_RW		0x3
+
+/*  define DECPROT lock */
+#define DECPROT_UNLOCK		0x0
+#define DECPROT_LOCK		0x1
+
+/* define TZMA IDs*/
+#define ETZPC_TZMA0_ID			200
+#define ETZPC_TZMA1_ID			201
+
+/* define ETZPC ID */
+#define STM32MP1_ETZPC_VREFBUF_ID	0
+#define STM32MP1_ETZPC_LPTIM2_ID	1
+#define STM32MP1_ETZPC_LPTIM3_ID	2
+#define STM32MP1_ETZPC_LTDC_ID		3
+#define STM32MP1_ETZPC_DCMIPP_ID	4
+#define STM32MP1_ETZPC_USBPHYCTRL_ID	5
+#define STM32MP1_ETZPC_DDRCTRLPHY_ID	6
+/* 7-11 Reserved */
+#define STM32MP1_ETZPC_IWDG1_ID		12
+#define STM32MP1_ETZPC_STGENC_ID	13
+/* 14-15 Reserved */
+#define STM32MP1_ETZPC_USART1_ID	16
+#define STM32MP1_ETZPC_USART2_ID	17
+#define STM32MP1_ETZPC_SPI4_ID		18
+#define STM32MP1_ETZPC_SPI5_ID		19
+#define STM32MP1_ETZPC_I2C3_ID		20
+#define STM32MP1_ETZPC_I2C4_ID		21
+#define STM32MP1_ETZPC_I2C5_ID		22
+#define STM32MP1_ETZPC_TIM12_ID		23
+#define STM32MP1_ETZPC_TIM13_ID		24
+#define STM32MP1_ETZPC_TIM14_ID		25
+#define STM32MP1_ETZPC_TIM15_ID		26
+#define STM32MP1_ETZPC_TIM16_ID		27
+#define STM32MP1_ETZPC_TIM17_ID		28
+/* 29-31 Reserved */
+#define STM32MP1_ETZPC_ADC1_ID		32
+#define STM32MP1_ETZPC_ADC2_ID		33
+#define STM32MP1_ETZPC_OTG_ID		34
+#define STM32MP1_ETZPC_TSC_ID		37
+/* 38-39 Reserved */
+#define STM32MP1_ETZPC_RNG_ID		40
+#define STM32MP1_ETZPC_HASH_ID		41
+#define STM32MP1_ETZPC_CRYP_ID		42
+#define STM32MP1_ETZPC_SAES_ID		43
+#define STM32MP1_ETZPC_PKA_ID		44
+#define STM32MP1_ETZPC_BKPSRAM_ID	45
+/* 46-47 Reserved */
+#define STM32MP1_ETZPC_ETH1_ID		48
+#define STM32MP1_ETZPC_ETH2_ID		49
+#define STM32MP1_ETZPC_SDMMC1_ID	50
+#define STM32MP1_ETZPC_SDMMC2_ID	51
+/* 52 Reserved */
+#define STM32MP1_ETZPC_MCE_ID		53
+#define STM32MP1_ETZPC_FMC_ID		54
+#define STM32MP1_ETZPC_QSPI_ID		55
+/* 56-59 Reserved */
+#define STM32MP1_ETZPC_SRAM1_ID		60
+#define STM32MP1_ETZPC_SRAM2_ID		61
+#define STM32MP1_ETZPC_SRAM3_ID		62
+/* 63 Reserved */
+
+#define STM32MP1_ETZPC_MAX_ID		64
+
+#define DECPROT(id, mode, lock)		(((id) << 16) | ((mode) << 8) | (lock))
+
+#endif /* _DT_BINDINGS_SOC_STM32MP13_ETZPC_H */
diff --git a/core/arch/arm/plat-stm32mp1/drivers/stm32mp1_etzpc.h b/core/include/dt-bindings/soc/stm32mp15-etzpc.h
similarity index 60%
rename from core/arch/arm/plat-stm32mp1/drivers/stm32mp1_etzpc.h
rename to core/include/dt-bindings/soc/stm32mp15-etzpc.h
index bc0b78fa2..ee6e9c917 100644
--- a/core/arch/arm/plat-stm32mp1/drivers/stm32mp1_etzpc.h
+++ b/core/include/dt-bindings/soc/stm32mp15-etzpc.h
@@ -1,13 +1,26 @@
-/* SPDX-License-Identifier: GPL-2.0 or BSD-3-Clause */
+/* SPDX-License-Identifier: BSD-2-Clause */
 /*
- * Copyright (C) 2018-2019, STMicroelectronics
+ * Copyright (c) 2018-2024, STMicroelectronics
  */
 
-#ifndef __STM32MP1_ETZPC_H
-#define __STM32MP1_ETZPC_H
+#ifndef _DT_BINDINGS_SOC_STM32MP15_ETZPC_H
+#define _DT_BINDINGS_SOC_STM32MP15_ETZPC_H
 
-/* Define DECPROT IDs for stm32mp1 familly */
-#ifdef CFG_STM32MP15
+/*  define DECPROT modes */
+#define DECPROT_S_RW		0x0
+#define DECPROT_NS_R_S_W	0x1
+#define DECPROT_MCU_ISOLATION	0x2
+#define DECPROT_NS_RW		0x3
+
+/*  define DECPROT lock */
+#define DECPROT_UNLOCK		0x0
+#define DECPROT_LOCK		0x1
+
+/* define TZMA IDs*/
+#define ETZPC_TZMA0_ID			200
+#define ETZPC_TZMA1_ID			201
+
+/* define ETZPC ID */
 #define STM32MP1_ETZPC_STGENC_ID	0
 #define STM32MP1_ETZPC_BKPSRAM_ID	1
 #define STM32MP1_ETZPC_IWDG1_ID		2
@@ -96,61 +109,9 @@
 #define STM32MP1_ETZPC_DLYBQ_ID		93
 #define STM32MP1_ETZPC_ETH_ID		94
 /* 95 Reserved */
+
 #define STM32MP1_ETZPC_MAX_ID		96
-#endif /* CFG_STM32MP15 */
 
-#ifdef CFG_STM32MP13
-#define STM32MP1_ETZPC_VREFBUF_ID	0
-#define STM32MP1_ETZPC_LPTIM2_ID	1
-#define STM32MP1_ETZPC_LPTIM3_ID	2
-#define STM32MP1_ETZPC_LTDC_ID		3
-#define STM32MP1_ETZPC_DCMIPP_ID	4
-#define STM32MP1_ETZPC_USBPHYCTRL_ID	5
-#define STM32MP1_ETZPC_DDRCTRLPHY_ID	6
-/* 7-11 Reserved */
-#define STM32MP1_ETZPC_IWDG1_ID		12
-#define STM32MP1_ETZPC_STGENC_ID	13
-/* 14-15 Reserved */
-#define STM32MP1_ETZPC_USART1_ID	16
-#define STM32MP1_ETZPC_USART2_ID	17
-#define STM32MP1_ETZPC_SPI4_ID		18
-#define STM32MP1_ETZPC_SPI5_ID		19
-#define STM32MP1_ETZPC_I2C3_ID		20
-#define STM32MP1_ETZPC_I2C4_ID		21
-#define STM32MP1_ETZPC_I2C5_ID		22
-#define STM32MP1_ETZPC_TIM12_ID		23
-#define STM32MP1_ETZPC_TIM13_ID		24
-#define STM32MP1_ETZPC_TIM14_ID		25
-#define STM32MP1_ETZPC_TIM15_ID		26
-#define STM32MP1_ETZPC_TIM16_ID		27
-#define STM32MP1_ETZPC_TIM17_ID		28
-/* 29-31 Reserved */
-#define STM32MP1_ETZPC_ADC1_ID		32
-#define STM32MP1_ETZPC_ADC2_ID		33
-#define STM32MP1_ETZPC_OTG_ID		34
-/* 35-36 Reserved */
-#define STM32MP1_ETZPC_TSC_ID		37
-/* 38-39 Reserved */
-#define STM32MP1_ETZPC_RNG_ID		40
-#define STM32MP1_ETZPC_HASH_ID		41
-#define STM32MP1_ETZPC_CRYP_ID		42
-#define STM32MP1_ETZPC_SAES_ID		43
-#define STM32MP1_ETZPC_PKA_ID		44
-#define STM32MP1_ETZPC_BKPSRAM_ID	45
-/* 46-47 Reserved */
-#define STM32MP1_ETZPC_ETH1_ID		48
-#define STM32MP1_ETZPC_ETH2_ID		49
-#define STM32MP1_ETZPC_SDMMC1_ID	50
-#define STM32MP1_ETZPC_SDMMC2_ID	51
-/* 52 Reserved */
-#define STM32MP1_ETZPC_MCE_ID		53
-#define STM32MP1_ETZPC_FMC_ID		54
-#define STM32MP1_ETZPC_QSPI_ID		55
-/* 56-59 Reserved */
-#define STM32MP1_ETZPC_SRAM1_ID		60
-#define STM32MP1_ETZPC_SRAM2_ID		61
-#define STM32MP1_ETZPC_SRAM3_ID		62
-/* 63 Reserved */
-#define STM32MP1_ETZPC_MAX_ID		64
-#endif /* CFG_STM32MP13 */
-#endif /*__STM32MP1_ETZPC_H*/
+#define DECPROT(id, mode, lock)		(((id) << 16) | ((mode) << 8) | (lock))
+
+#endif /* _DT_BINDINGS_SOC_STM32MP15_ETZPC_H */
-- 
2.47.2

